### DISCUSSION TITLE
Firing extension actions for a pre-existing controller

### DISCUSSION THREAD

**Discussion by huffstler** [FreshRSS/Extensions#258] [Q&A]
Hey all, this is a follow up to the discussion I [opened here](https://github.com/FreshRSS/FreshRSS/discussions/6878).

I found some free time this morning and have been going through the code for the [Pocket-Button](https://github.com/christian-putzke/freshrss-pocket-button) Extension.

From what I've gathered from [the docs (#Routing Section)](https://freshrss.github.io/FreshRSS/en/developers/Minz/), what I want is to kick off the [`addAction`](https://github.com/christian-putzke/freshrss-pocket-button/blob/main/Controllers/pocketButtonController.php#L94) when the star button is clicked. (After pocket auth has happened)

I think that in order to do that, I would need to call `Minz_Request::forward()` from my handle function defined in the `extension.php`. With that being said, I also saw this in the docs: 

> Note: it is preferable to use Minz_Request only in controllers. It is likely that you will encounter this method in FreshRSS views, or even in templates, but be aware that this is not good practice.

So if my first idea is not recommended, what would be the convention for doing something like this?

Unfortunately, I'm hit with the double whammy of 1. Not knowing PHP, and 2. Not having much experience at front end stuff. It's all very new and confusing right now. 😵‍💫

**Comment by huffstler** [comment]
Here's the repo that I've set up for the extension if you want something to look through

https://github.com/huffstler/FreshRSS-StarToPocket

**Comment by huffstler** [comment]
This is actually done now, the extension is made and works currently. There's a side bug that I found where star actions that happen through the greader api aren't firing the hooks that I would have expected. Once that bug is fixed, the extension _should_ work as intended.

**Reply by Frenzie** [reply]
Maybe @Alkarex knows something about it.

**Reply by Alkarex** [reply]
Hello,
It is the same question than what is discussed in https://github.com/FreshRSS/FreshRSS/discussions/6878#discussioncomment-11354539 , right?

**Reply by huffstler** [reply]
Hey, yes it is. I was a little confused about where the conversation should happen for my original question and subsequent debugging questions. My apologies for the double threads.

**Reply by Alkarex** [reply]
No worries
